src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java

```diff
@@ -316,13 +316,18 @@ public Response postProgressReport(
 
     for (Integer datasetId : childDar.getDatasetIds()) {
       Dataset dataset = datasetService.findDatasetById(datasetId);
+      if (dataset == null) {
+        throw new NotFoundException("Dataset " + datasetId + " not found");
```

Comment:
Another pattern we often use is to throw a NFE in the service class. It doesn't look like we do that here though.

Reply:
Yes, I noticed that in the other services classes. I'll update the code to do that.

Reply:
After taking a deeper look at this it would be non-trivial work that would have side effects on other code we've already written. I think I'll just create a tech debt ticket instead.